tests/lib/rules/sort-comp.js

```diff
@@ -504,6 +504,16 @@ ruleTester.run('sort-comp', rule, {
     ].join('\n'),
     parser: 'babel-eslint',
     errors: [{message: 'render should be placed after displayName'}]
+  }, {
+    // Must validate static lifecycle methods
+    code: [
+      'class Hello extends React.Component {',
+      '  static getDerivedStateFromProps() {}',
+      '  constructor() {}',
+      '}'
+    ].join('\n'),
+    parser: 'babel-eslint',
```

Comment:
the default parser should be sufficient here; please either add a duplicate test (one using default, one using babel-eslint) or, since this is all ES6 syntax, just have the one using the default parser.